src/libinjector/win_injector.c

```diff
@@ -948,12 +950,25 @@ static event_response_t injector_int3_cb(drakvuf_t drakvuf, drakvuf_trap_info_t*
     if (info->regs->rip != info->trap->breakpoint.addr)
         return 0;
 
+    // on multi-vCPU systems, multiple injector_int3_cb might be active at the same time
+    // so we do need thread safety here
```

Comment:
That's not how it currently works, events are always read 1-by-1 from the ring buffer by LibVMI. Even if the guest has multiple vCPUs and both trap only one event is processed from the ring at the time.

Reply:
hm, so simple check for tid=0 will be enough, no need for thread safety